storage/remote/codec.go

```diff
@@ -167,6 +167,122 @@ func FromQueryResult(sortSeries bool, res *prompb.QueryResult) storage.SeriesSet
 	}
 }
 
+// design approach taken from https://prometheus.io/blog/#solution
+// chunkedResponseSeriesSet implements storage.SeriesSet.
+type chunkedResponseSeriesSet struct {
+	stream        *ChunkedReader
+	currentChunks []prompb.Chunk
+	currentLabels []prompb.Label
+	closer        io.Closer
+	responseError error
+}
+
+func (c *chunkedResponseSeriesSet) Next() bool {
+	c.currentLabels = nil
+	c.currentChunks = nil
+	res := &prompb.ChunkedSeries{}
```

Comment:
i think this should be use prompb.ChunkedReadResponse type for unmarshal, like [thanos](https://github.com/thanos-io/thanos/blob/c733564d44745af1a023bfa5d51d6d205404dc82/pkg/store/prometheus.go#L285)

Reply:
Hmm.
WDYT @bwplotka ?

Reply:
It's kind of easy to say. Just check what is Written by remote read handler (: 

And it's here: https://github.com/prometheus/prometheus/blob/b788986717e1597452ca25e5219510bb787165c7/storage/remote/codec.go#L359

so hope that answers your question (:

Reply:
Yes. Thanks for the help/guidance Bartek.